### DISCUSSION TITLE
Please add summary of failed tests at end of `go test`

### DISCUSSION THREAD

**Discussion by jiridanek** [skupperproject/skupper#737] [Ideas]
It is difficult to look for failed tests in `go test` output. Grepping for `FAIL` in webbrowser is hard if search does not have case-sensitive option and if the word appears in different contexts elsewhere in the output.

Go issue about printing final FAIL, not very helpful here https://github.com/golang/go/issues/30507

Tool to postprocess the `go test` output, https://github.com/jstemmer/go-junit-report
Links another tool, `richgo`, https://github.com/sclevine/spec/issues/9

**Comment by jiridanek** [comment]
After reading through more issues, found also

https://github.com/gotestyourself/gotestsum (this looks good)

https://stackoverflow.com/questions/54942550/how-to-easily-find-out-which-tests-fail

suggestion to use --json output format of go test

**Comment by jiridanek** [comment]
gotestsum practical example https://github.com/jiridanek/skupper-router/runs/6198578904?check_suite_focus=true#step:4:1929

**Comment by fgiorgetti** [comment]
We also use go-junit-report since CircleCI also provides an interface for displaying test results, see:

https://circleci.com/blog/how-to-output-junit-tests-through-circleci-2-0-for-expanded-insights/?utm_source=google&utm_medium=sem&utm_campaign=sem-google-dg--latam-en-dsa-maxConv-auth-nb&utm_term=g_-_c__dsa_&utm_content=&gclid=Cj0KCQjw06OTBhC_ARIsAAU1yOWnot2LVYb0KkEnJ3hXE-ECZceMbDIt3bWyDDu0RMBpNQoskNunYwMaAlJlEALw_wcB

**Comment by jiridanek** [comment]
There seems to be a way to get this in GHA as well, I'm looking at screenshots at https://github.com/marketplace/actions/test-reporter
